apps/taxes/src/modules/taxjar/taxjar-calculate-taxes-adapter.test.ts

```diff
@@ -0,0 +1,148 @@
+import { describe, expect, it } from "vitest";
+import { TaxBaseFragment } from "../../../generated/graphql";
+import { ChannelConfig } from "../channels-configuration/channels-config";
+import { TaxJarCalculateTaxesPayloadTransformer } from "./taxjar-calculate-taxes-adapter";
+
+const MOCKED_TAX_BASE: TaxBaseFragment = {
+  pricesEnteredWithTax: false,
+  currency: "PLN",
+  channel: {
+    slug: "channel-pln",
+  },
+  sourceObject: {
+    __typename: "Order",
+    user: {
+      id: "VXNlcjo5ZjY3ZjY0Zi1iZjY5LTQ5ZjYtYjQ4Zi1iZjY3ZjY0ZjY0ZjY=",
+    },
+  },
+  discounts: [],
+  address: {
+    streetAddress1: "123 Palm Grove Ln",
+    streetAddress2: "",
+    city: "LOS ANGELES",
+    country: {
+      code: "US",
+    },
+    countryArea: "CA",
+    postalCode: "90002",
+  },
+  shippingPrice: {
+    amount: 48.33,
+  },
+  lines: [
+    {
+      quantity: 3,
+      unitPrice: {
+        amount: 84,
+      },
+      totalPrice: {
+        amount: 252,
+      },
+      sourceLine: {
+        __typename: "OrderLine",
+        id: "T3JkZXJMaW5lOmY1NGQ1MWY2LTc1OTctNGY2OC1hNDk0LTFjYjZlYjRmOTlhMQ==",
+        variant: {
+          id: "UHJvZHVjdFZhcmlhbnQ6MzQ2",
+          product: {
+            metafield: null,
+            productType: {
+              metafield: null,
+            },
+          },
+        },
+      },
+    },
+    {
+      quantity: 1,
+      unitPrice: {
+        amount: 5.99,
+      },
+      totalPrice: {
+        amount: 5.99,
+      },
+      sourceLine: {
+        __typename: "OrderLine",
+        id: "T3JkZXJMaW5lOjU1NTFjNTFjLTM5MWQtNGI0Ny04MGU0LWVjY2Q5ZjU4MjQyNQ==",
+        variant: {
+          id: "UHJvZHVjdFZhcmlhbnQ6Mzg1",
+          product: {
+            metafield: null,
+            productType: {
+              metafield: null,
+            },
+          },
+        },
+      },
+    },
+  ],
+};
+
+const MOCKED_CHANNEL: ChannelConfig = {
+  providerInstanceId: "b8c29f49-7cae-4762-8458-e9a27eb83081",
+  enabled: false,
+  address: {
+    country: "US",
+    zip: "92093",
+    state: "CA",
+    city: "La Jolla",
+    street: "9500 Gilman Drive",
+  },
+};
+
+/*
+ * const MOCKED_TAXJAR_CONFIG = {
+ *   apiKey: "5da2f821eee4035db4771edab942a4cc",
+ *   isSandbox: true,
+ *   name: "taxjar-1",
+ * };
+ */
+
+describe("TaxJarCalculateTaxesPayloadTransformer", () => {
+  it("should return payload containing line_items", () => {
```

Comment:
we need to have tests that will allow many payloads. You can create a `getMockedEvent()` method, later you can have scenario like 

pseudocode:
```
"Maps empty lines"

const payload = getMockedPayload();
payload.items = [] 

expect(...)
```

We need various scenarios for diverse set of payloads

Reply:
Sorry, I don't follow. What would `getMockedPayload/Event` method do?

Reply:
When would I mutate the payload during the test? Can you give a more descriptive example of what you mean?

Reply:
I added `taxjar-mock-factory.ts`, please have a look